### ISSUE TITLE
computing camera poses 

### ISSUE THREAD

**Issue by takuphilchan** [lyx0208/3dSwap#7]
Hello!
I am trying to compute camera poses for custom images. I used eg3d to crop the images, can you please explain how I can compute camera poses for my custom images. I want to produce the corresponding numpy camera pose file for each image like you did with ffhq dataset. 

Thank you.

**Comment by lyx0208** [comment]
Hello!
Did you get a "cameras.json" file when you cropped the custom images with eg3d? If so, you can just ignore the "intrinsic" insight it, and concat the fixed intrinsic "[4.2647, 0.0, 0.5, 0.0, 4.2647, 0.5, 0.0, 0.0, 1.0]" with "pose" in it. Then you can get the 1*25 array as you want.

**Comment by takuphilchan** [comment]
Thank you for your reply. Can you please elaborate on the above mentioned "concatenation process" after I get cameras.json and cropped images. Is there a file I can use to perform this process or its a manual process? 

> Hello! Did you get a "cameras.json" file when you cropped the custom images with eg3d? If so, you can just ignore the "intrinsic" insight it, and concat the fixed intrinsic "[4.2647, 0.0, 0.5, 0.0, 4.2647, 0.5, 0.0, 0.0, 1.0]" with "pose" in it. Then you can get the 1*25 array as you want.

**Comment by lyx0208** [comment]
I just write a script:

import json
import numpy as np
import torch
if __name__ == '__main__':
    with open('custom/cameras.json', 'r') as f:
        data = json.load(f)
    for i in range(list(data.keys()).__len__()):
        name = list(data.keys())[i].split('.')[0]
        pose = torch.tensor(data[list(data.keys())[i]]['pose']).view([1, 16])
        intrinsics = torch.tensor([[4.2647, 0.0, 0.5, 0.0, 4.2647, 0.5, 0.0, 0.0, 1.0]])
        p = torch.cat([pose, intrinsics], dim=1)[0]
        print(p)
        np.save('custom/' + name + '.npy', p)
        print(name)

**Comment by takuphilchan** [comment]
Okay, thank you.
